### PR TITLE
v1

### PR DESCRIPTION
**Pull request by zhirzh**
## added
- `SingleTapButton` - debounced `Button` with fixed interval of 2000ms

### `DateTimePicker`
- allow user to cancel date selection
- restore support for "type" prop

### `SearchInput`
- publicly expose `closeModal` method

### `Login` recipe
- render Login screen footer as render prop `getFooter`
- allow user to submit OTP only once

### CODE REVIEW

File: src/components/Input.tsx [hunk 1]
```diff
@@ -86,7 +86,8 @@ class Input extends React.PureComponent<InputProps, InputState> {
       () => {
         Animated.timing(this.state.placeholderMarginTop, {
           toValue: 0,
-          duration: 200
+          duration: 200,
+          useNativeDriver: false
```
Comment (l3satwik) [inline src/components/Input.tsx:90]: shouldn't we prefer using native driver?

File: src/components/SingleTapButton.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  };
+  private onPress: ButtonProps["onPress"] = debounce(
+    (e: GestureResponderEvent) => {
+      this.setState({ loading: true });
+      // Using function form of finally as want to error out if not returned a Promise
+      this.props.onPress(e).finally(() => {
+        this.setState({ loading: false });
+      });
+    },
+    2000,
```
Comment (l3satwik) [inline src/components/SingleTapButton.tsx:30]: should be set from prop with default value.

**Review by l3satwik — APPROVED** [review]

File: src/components/Input.tsx (hunk 1 above)
Comment (zhirzh) [reply src/components/Input.tsx:90]: https://reactnative.dev/blog/2017/02/14/using-native-driver-for-animated#caveats

File: src/components/SingleTapButton.tsx (hunk 2 above)
Comment (zhirzh) [reply src/components/SingleTapButton.tsx]: fixed in e41152e

File: package.json [hunk 3]
```diff
@@ -14,44 +15,43 @@
     "prepublishOnly": "yarn build:dist"
   },
   "dependencies": {
+    "@react-native-community/datetimepicker": "^2.3.2",
     "date-fns": "^1.30.1",
```
[the remark this answers is not in the document]
Comment (zhirzh) [reply package.json:19]: expo 37 (and 38) don't support ~v2~ v3 yet
https://github.com/expo/expo/blob/master/CHANGELOG.md#3700

[the remark this answers is not in the document]
Comment (zhirzh) [reply package.json:19]: I think you mean v3

anyhow, expo 37 (and 38) only support v2
https://github.com/expo/expo/blob/master/CHANGELOG.md#3700

File: app.json [hunk 4]
```diff
@@ -2,9 +2,9 @@
   "expo": {
     "name": "pebble-native",
     "description": "A set of components developed for React Native.",
-    "slug": "typescript",
+    "slug": "pebble-native",
     "privacy": "public",
-    "sdkVersion": "32.0.0",
+    "sdkVersion": "37.0.0",
```
[the remark this answers is not in the document]
Comment (zhirzh) [reply app.json:7]: expo is only used for pebble-native demo app

**Pull request merged**